### PR TITLE
Adding variable support to the mana cost condition

### PR DESCRIPTION
**Pull request by jhallbachner**
This adds support for variables in the mana search, inspired by the internal R&D code letters Mark Rosewater has mentioned in the past. "{m}," "{n}," and "{o}" each match any single color of mana, while "{h}" matches any single hybrid symbol. They can be combined, so "mana={m}{n}{o}" will find any card with a cost of exactly one of three different colors of mana, while "mana>={m}{h}" will find every card with at least one colored mana and one hybrid mana in its cost.

This also includes tests to verify this functionality and updates the syntax page of the frontend app to describe how it works.

### CODE REVIEW

File: frontend/app/views/help/syntax.html.haml [hunk 1]
```diff
@@ -221,4 +232,4 @@
       = search_help "sort:new", "sort results by most recent first, counting only regular sets for sake of order (other printings will go at end of results)"
       = search_help "sort:newall", "sort results by most recent first, counting all printings"
       = search_help "sort:ord", "sort results by oldest first, counting only regular sets for sake of order"
-      = search_help "sort:ordall", "sort results by oldest first, counting all printings"
+      = search_help "sort:ordall", "sort results by oldest first, counting all printings"
```
Comment (taw) [inline frontend/app/views/help/syntax.html.haml]: It seems that your editor deletes final newline here. This is going to cause unnecessary conflicts in git patches.

**Comment by taw** [comment]
It's a good idea.

I'll just have to double check the code as it's a bit complex and probably has a lot of edge cases.

**Comment by taw** [comment]
We'd need to decide (and obviously test) some edge cases:

* presumably {m}{n} is the same {w}{u} or {w}{b} or ..  (20 cases, 2 duplicate)
* how about {w}{m} ? does that include {w}{w} or only 4 other combinations? My intuition is that it wouldn't include {w}{w}.
* depending on previous answer, how about {m}{h}, {m}{u/b} etc. My intuition is that it such case {m} could be any of 5 colors.

**Comment by taw** [comment]
Also a fun story, it currently conflicts with this use of {h}: http://mtg.wtf/card?q=mana%3D%7Bhw%7D

**Comment by taw** [comment]
This is now in master with some extra tests and minor edits.
